eclipse-collections/src/main/java/org/eclipse/collections/impl/stack/immutable/ImmutableArrayStack.java

```diff
@@ -112,7 +112,11 @@
 
 /**
  * The immutable equivalent of ArrayStack. Wraps a FastList.
+ *
+ * Deprecated this class because it was replaced by singly-linked implementation of immutable stack.
+ * https://github.com/eclipse/eclipse-collections/pull/767
```

Comment:
No need to have a PR here. You should have a reference of what it was replaced by as a link to code.
Take a look at other `deprecate` examples in the project.